php/getsettings.php

```diff
@@ -2,19 +2,6 @@
 
 require_once( 'util.php' );
 
-$s = '{}';
-$fname = getSettingsPath()."/uisettings.json";
-$fo = @fopen($fname, 'r');
-if($fo!==false)
-{
-	if(flock($fo, LOCK_SH))
-	{
-		$s = @file_get_contents($fname);
-		if($s==false)
-			$s = '{}';
-		flock($fo, LOCK_UN); 
-	}
-	fclose($fo); 
-}
-
-cachedEcho($s,"application/json",true);
+$fname = getWebUIJsonFile()."/uisettings.json";
```

Comment:
What about race conditions (when we have reading and writing of this file at the same time), how you will prevent them?

Reply:
I will prevent writing to this file until it has been read by the WebUI. Then after it has been read, will attempt to write any changes. `theWebUI.save()` is the only function which calls `?action=setuisettings` where it writes inside `setsettings.php`.

Reply:
I'm going to rewrite some code. I will use the rCache class to store the webui settings in a .dat file. I will let this class handle the file locking. There will be no race conditions when I'm done. The WebUI will still be able to make async requests.